lib/index.js

```diff
@@ -191,12 +191,16 @@ CloudFront.prototype.request = function(method, path, headers, cb) {
 
       cb(err, info);
     });
-    parser.on("error", function(err) {
-      cb(err, info);
+    parser.on('error', function(err) {
+      cb(err);
     });
 
   });
 
+  req.on('error', function(err) {
```

Comment:
If I remember correctly it is possible to receive an `error` event after the `response` event. We should probably add some code to handle that. Otherwise, this is better than not having this.